pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -492,7 +490,7 @@ static int NpyArr_iterNextItem(JSOBJ obj, JSONTypeContext *tc) {
     ((PyObjectEncoder *)tc->encoder)->npyValue = npyarr->dataptr;
     ((PyObjectEncoder *)tc->encoder)->npyCtxtPassthru = npyarr;
   } else {
-    GET_TC(tc)->itemValue = npyarr->getitem(npyarr->dataptr, npyarr->array);
+    GET_TC(tc)->itemValue = PyArray_GETITEM(npyarr->array, npyarr->dataptr);
```

Comment:
The failing build doesn't like the implicit conversion from a PyArrayObject* to a PyObject* . Should be able to type-check / cast to make the compiler happy